File: on_thi_chuyen/apple_division.py
n = int(input())
a = list(map(int, input().split()))
total = sum(a)
ans = 10**18
def dfs(i, sum1):
    global ans
    if i == n:
        sum2 = total - sum1
        ans = min(ans, abs(sum1 - sum2))
        return
    dfs(i + 1, sum1 + a[i])
    dfs(i + 1, sum1)
dfs(0, 0)
print(ans)

File: c_plus_plus/apple_division.cpp
#include <iostream>
#include <cmath>
using namespace std;

long long n,min_dif,total;
long long arr[20];

void dfs(int i,long long sum1){
    if(i == n){
        long long sum2 = total - sum1;
        long long dif = abs(sum1 - sum2);
        if(dif < min_dif){
            min_dif = dif;
        }
        return;
    }

    dfs(i + 1,sum1 + arr[i]);
    dfs(i + 1,sum1);
}

int main(){
    cin >> n;
    for(int i = 0;i < n;i++){
        cin >> arr[i];
        total += arr[i];
    }
    min_dif = total;

    dfs(0,0);
    cout << min_dif;

}
